File: packages/luciq_flutter/android/src/main/java/ai/luciq/flutter/modules/LuciqLogApi.java
package ai.luciq.flutter.modules;

import androidx.annotation.NonNull;

import ai.luciq.flutter.generated.LuciqLogPigeon;
import ai.luciq.flutter.util.LuciqFlutterDebugTags;
import ai.luciq.flutter.util.LuciqFlutterLogger;
import ai.luciq.library.logging.LuciqLog;

import io.flutter.plugin.common.BinaryMessenger;

public class LuciqLogApi implements LuciqLogPigeon.LuciqLogHostApi {
    public static void init(BinaryMessenger messenger) {
        final LuciqLogApi api = new LuciqLogApi();
        LuciqLogPigeon.LuciqLogHostApi.setup(messenger, api);
    }

    @Override
    public void logVerbose(@NonNull String message) {
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.logVerbose] phase=enter length=" + message.length());
        LuciqLog.v(message);
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.logVerbose] phase=exit");
    }

    @Override
    public void logDebug(@NonNull String message) {
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.logDebug] phase=enter length=" + message.length());
        LuciqLog.d(message);
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.logDebug] phase=exit");
    }

    @Override
    public void logInfo(@NonNull String message) {
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.logInfo] phase=enter length=" + message.length());
        LuciqLog.i(message);
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.logInfo] phase=exit");
    }

    @Override
    public void logWarn(@NonNull String message) {
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.logWarn] phase=enter length=" + message.length());
        LuciqLog.w(message);
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.logWarn] phase=exit");
    }

    @Override
    public void logError(@NonNull String message) {
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.logError] phase=enter length=" + message.length());
        LuciqLog.e(message);
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.logError] phase=exit");
    }

    @Override
    public void clearAllLogs() {
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.clearAllLogs] phase=enter");
        LuciqLog.clearLogs();
        LuciqFlutterLogger.d(LuciqFlutterDebugTags.LUCIQ_LOG,
                "[LOG.clearAllLogs] phase=exit");
    }
}

File: packages/luciq_flutter/ios/Classes/Modules/LuciqLogApi.m
#import "LuciqSDK/LuciqSDK.h"
#import "LuciqLogApi.h"
#import "../Util/LuciqFlutterLogger.h"
#import "../Util/LuciqFlutterDebugTags.h"

extern void InitLuciqLogApi(id<FlutterBinaryMessenger> messenger) {
    LuciqLogApi *api = [[LuciqLogApi alloc] init];
    LuciqLogHostApiSetup(messenger, api);
}

@implementation LuciqLogApi

- (void)logVerboseMessage:(NSString *)message error:(FlutterError *_Nullable *_Nonnull)error {
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.logVerbose] phase=enter length=%lu",
        (unsigned long)message.length];
    [LCQLog logVerbose:message];
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.logVerbose] phase=exit"];
}

- (void)logDebugMessage:(NSString *)message error:(FlutterError *_Nullable *_Nonnull)error {
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.logDebug] phase=enter length=%lu",
        (unsigned long)message.length];
    [LCQLog logDebug:message];
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.logDebug] phase=exit"];
}

- (void)logInfoMessage:(NSString *)message error:(FlutterError *_Nullable *_Nonnull)error {
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.logInfo] phase=enter length=%lu",
        (unsigned long)message.length];
    [LCQLog logInfo:message];
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.logInfo] phase=exit"];
}

- (void)logWarnMessage:(NSString *)message error:(FlutterError *_Nullable *_Nonnull)error {
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.logWarn] phase=enter length=%lu",
        (unsigned long)message.length];
    [LCQLog logWarn:message];
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.logWarn] phase=exit"];
}

- (void)logErrorMessage:(NSString *)message error:(FlutterError *_Nullable *_Nonnull)error {
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.logError] phase=enter length=%lu",
        (unsigned long)message.length];
    [LCQLog logError:message];
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.logError] phase=exit"];
}

- (void)clearAllLogsWithError:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.clearAllLogs] phase=enter"];
    [LCQLog clearAllLogs];
    [LuciqFlutterLogger d:[LuciqFlutterDebugTags luciqLog]
                   format:@"[LOG.clearAllLogs] phase=exit"];
}

@end
